__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,888 @@
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // @ts-ignore - for private method
+            vm.aggregateTransactionDetails = transaction;
+
+            // Act:
+            vm.needsCosignature();
+
+            // Assert:
+            // @ts-ignore - for private method
+            expect(vm.transactionSigningFlag).toBe(expectedResult);
+        };
+
+        test('set transactionSigningFlag to false when current account is multisig', () => {
+            // Arrange:
+            const cosig1Address = 'TALPBVKED63OTOS6LNKFIE4H357MBOQPVGJBLOI';
+            const cosig2Address = 'TD2KY5BKECF6YKSWSTGMUHFP66GG33S5MG6UOYQ';
+
+            const multisigAccountInfo = new MultisigAccountInfo(
+                1,
+                currentSigner.address,
+                2,
+                1,
+                [Address.createFromRawAddress(cosig1Address), Address.createFromRawAddress(cosig2Address)],
+                [],
+            );
+
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile, currentAccount: WalletsModel1, currentAccountMultisigInfo: multisigAccountInfo },
+                { transaction: mockAggregateTransaction() },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            vm.needsCosignature();
+
+            // Assert:
+            // @ts-ignore - for private method
+            expect(vm.transactionSigningFlag).toBe(false);
+        });
+
+        test('set transactionSigningFlag to false when non condition meet', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile, currentAccount: WalletsModel1 },
+                { transaction: mockAggregateTransaction() },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            vm.needsCosignature();
+
+            // Assert:
+            // @ts-ignore - for private method
+            expect(vm.transactionSigningFlag).toBe(false);
+        });
+
+        test('set transactionSigningFlag to false when transaction signed by current account', () => {
+            // Arrange:
+            const aggregateTransaction = mockAggregateTransaction(
+                [mockTransferTransaction()],
+                [
+                    new AggregateTransactionCosignature(
+                        '939673209A13FF82397578D22CC96EB8516A6760C894D9B7535E3A1E068007B9255CFA9A914C97142A7AE18533E381C846B69D2AE0D60D1DC8A55AD120E2B606',
+                        PublicAccount.createFromPublicKey(WalletsModel1.publicKey, NetworkType.TEST_NET),
+                        UInt64.fromUint(0),
+                    ),
+                ],
+            );
+
+            runBasicNeedsCosignatureTests(aggregateTransaction, false);
+        });
+
+        test('set transactionSigningFlag to false when cosignerAddresses excluded current account', () => {
```

Comment:
what do you mean by "excluded" here and in next test ?

Reply:
basically, the transaction is not required for the current account.
let me update the description